packages/server-wallet/src/wallet/wallet.ts

```diff
@@ -224,13 +256,16 @@ export class Wallet extends EventEmitter<WalletEvents> {
   public async closeChannels(channelIds: string[]): Promise<ObjectiveResult[]> {
     return Promise.all(
       channelIds.map(async channelId => {
+        const objectiveId = ['CloseChannel', channelId].join('-');
```

Comment:
```suggestion
        const objectiveId = `CloseChannel-${channelId}`;
```
I personally have to think less reading this than the join. I dug around a bit here to see what would be involved in type-checking the `'CloseChannel'` string, and it seemed surprisingly complicated! The string literal union type of `_Objective.type` is inferred from the literals supplied in implementing generic types.

I have no comment on whether this is an anti-pattern, but it took a few minutes to wrap my head around it.

Reply:
I ended up [updating](https://github.com/statechannels/statechannels/pull/3582/commits/9fa21b60972dedd68423a476293c7cdfa1e048fd) `wallet-core` to expose a `calculateObjectiveId` which can generate an objective id with just the type and channel Id.

That way the `wallet` doesn't have to worry at all about how an objective id is calculated.